[CI] RemoteClusterSecurityEsqlIT testCrossClusterEnrichWithOnlyRemotePrivs failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10258 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/7nlwk2akoqm5w)
- [elasticsearch-periodic-platform-support #10258 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/gp3re4br56y7q)
- [elasticsearch-periodic-platform-support #10258 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/nwreexc3letsu)
- [elasticsearch-periodic-platform-support #10251 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/fttyruslhekbc)
- [elasticsearch-periodic-platform-support #10251 / opensuse-leap-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/yu3q44jwa4xcc)
- [elasticsearch-periodic-platform-support #10237 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/wut2xs5x3dp5w)
- [elasticsearch-periodic-platform-support #10237 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/lak4mgvtj734k)
- [elasticsearch-periodic-platform-support #10230 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/m4iivs4d52ims)
- [elasticsearch-periodic-platform-support #10209 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/qdsstat5blz5a)
- [elasticsearch-periodic #10210 / openjdk23_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/e7idjuau7wsji)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:security:qa:multi-cluster:javaRestTest" --tests "org.elasticsearch.xpack.remotecluster.RemoteClusterSecurityEsqlIT.testCrossClusterEnrichWithOnlyRemotePrivs" -Dtests.seed=F57E11F840B8ECBA -Dtests.locale=ko -Dtests.timezone=America/Argentina/Salta -Druntime.java=23
```

**Applicable branches:**
8.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.remotecluster.RemoteClusterSecurityEsqlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCrossClusterEnrichWithOnlyRemotePrivs'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: iterable with items [<1>, <3>, "usa", "germany"] in any order
     but: not matched: "japan"
```

**Issue Reasons:**
- [8.17] 3 consecutive failures in test testCrossClusterEnrichWithOnlyRemotePrivs
- [8.17] 16 failures in test testCrossClusterEnrichWithOnlyRemotePrivs (5.5% fail rate in 289 executions)
- [8.17] 2 failures in step rocky-9_platform-support-unix (25.0% fail rate in 8 executions)
- [8.17] 3 failures in step oraclelinux-8_platform-support-unix (33.3% fail rate in 9 executions)
- [8.17] 2 failures in step sles-15_platform-support-unix (22.2% fail rate in 9 executions)
- [8.17] 2 failures in step ubuntu-2204_platform-support-unix (22.2% fail rate in 9 executions)
- [8.17] 7 failures in pipeline elasticsearch-periodic-platform-support (70.0% fail rate in 10 executions)
- [8.17] 3 failures in pipeline elasticsearch-periodic (30.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityEsqlIT.java
+++ b/x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityEsqlIT.java
@@ -797,18 +797,18 @@ public class RemoteClusterSecurityEsqlIT extends AbstractRemoteClusterSecurityTe
             | ENRICH countries
             | STATS size=count(*) by country
             | SORT size DESC
-            | LIMIT 2"""));
+            | LIMIT 3"""));
         assertOK(response);
 
         Map<String, Object> responseAsMap = entityAsMap(response);
         List<?> columns = (List<?>) responseAsMap.get("columns");
         List<?> values = (List<?>) responseAsMap.get("values");
         assertEquals(2, columns.size());
-        assertEquals(2, values.size());
+        assertEquals(3, values.size());
         List<?> flatList = values.stream()
             .flatMap(innerList -> innerList instanceof List ? ((List<?>) innerList).stream() : Stream.empty())
             .collect(Collectors.toList());
-        assertThat(flatList, containsInAnyOrder(1, 3, "usa", "germany"));
+        assertThat(flatList, containsInAnyOrder(1, 1, 3, "usa", "germany", "japan"));
     }
 
     private void createAliases() throws Exception {

